contracts/MultisigCaller.sol

```diff
@@ -0,0 +1,285 @@
+
+        requiredApprovals = _requiredApprovals;
+
+        // Setup the contract itself as the admin
+        _grantRole(DEFAULT_ADMIN_ROLE, address(this));
+        _grantRole(ADMIN_ROLE, address(this));
+
+        // Setup initial approvers
+        for (uint256 i = 0; i < approvers.length; i++) {
+            if (approvers[i] == address(0)) revert ZeroAddress();
+            _grantRole(APPROVER_ROLE, approvers[i]);
+        }
+    }
+
+    function submitTransaction(address to, uint256 value, bytes memory data) external payable nonReentrant onlyRole(APPROVER_ROLE) {
+        uint256 txId = transactions.length;
+        
+        Transaction memory newTx = Transaction({
+            to: to,
+            value: value,
+            data: data,
+            executed: false,
+            approvalCount: 0
+        });
+        transactions.push(newTx);
+        
+        emit TransactionSubmitted(txId, to, value, data);
+
+        Transaction storage transaction = transactions[txId];
+        _approveTransaction(transaction, txId);
+    }
+
+    function approveTransaction(uint256 txId) public payable nonReentrant onlyRole(APPROVER_ROLE) {
+        if (txId >= transactions.length) revert InvalidTransactionId(txId);
+        Transaction storage transaction = transactions[txId];
+        _approveTransaction(transaction, txId);
+    }
+
+    function _approveTransaction(Transaction storage transaction, uint256 txId) private {
+        if (approvals[txId][msg.sender]) revert TransactionAlreadyApproved(txId, msg.sender);
+
+        approvals[txId][msg.sender] = true;
+        transaction.approvalCount++;
+
+        emit TransactionApproved(txId, msg.sender);
+
+        if (transaction.approvalCount >= requiredApprovals) {
+            _executeTransaction(transaction, txId);
+        }
+    }
+
+    function _executeTransaction(Transaction storage transaction, uint256 txId) internal {
+        if (transaction.executed) revert TransactionAlreadyExecuted(txId);
+        
+        transaction.executed = true;
+        
+        (bool success, bytes memory returndata) = transaction.to.call{value: transaction.value}(
+            transaction.data
+        );
+
+        Address.verifyCallResult(success, returndata);
+        
+        emit TransactionExecuted(txId);
+    }
+
+    function addApprover(address account) public onlyRole(ADMIN_ROLE) {
+        uint256 currentCount = getRoleMemberCount(APPROVER_ROLE);
+        if (currentCount >= MAX_APPROVERS) revert MaxApproversReached(MAX_APPROVERS);
+        grantRole(APPROVER_ROLE, account);
+        emit ApproverAdded(account);
+    }
+
+    function removeApprover(address account) public onlyRole(ADMIN_ROLE) {
+        uint256 currentCount = getRoleMemberCount(APPROVER_ROLE);
+        if (currentCount <= requiredApprovals) 
+            revert InsufficientApprovers(requiredApprovals, currentCount);
+        revokeRole(APPROVER_ROLE, account);
```

Comment:
same here as in `addApprover` function

Reply:
Done in ee247cf8db19dc8ac7894135cae2bea43027a82b.